### PR TITLE
[WIP] fixes case where dangling loans in OL database by checking waitlist d…

### PR DESCRIPTION
**Pull request by mekarpeles**
…irectly from IA avail v2 API

### CODE REVIEW

**Comment by mekarpeles** [comment]
@hornc suggestions how we might go about testing this?

File: openlibrary/core/models.py [hunk 1]
```diff
@@ -275,17 +275,25 @@ def is_in_private_collection(self):
...
+        return self.in_borrowable_collection()
+
     def get_waitinglist(self):
         """Returns list of records for all users currently waiting for this book."""
         return waitinglist.get_waitinglist_for_book(self.key)
 
-    def get_waitinglist_size(self):
+    def get_ia_waitlist_size(self):
+        availability = lending.get_availability_of_ocaid(self.get('ocaid'))
+        return int(availability.get('responses', {}).get(self.get('ocaid'), {}).get('num_waitlist', 0))
```
Comment (hornc) [inline openlibrary/core/models.py]: this return looks like its doing a lot of work, maybe `lending` should handle the logic for extracting the waitlist size?

File: openlibrary/core/waitinglist.py [hunk 2]
```diff
@@ -189,9 +189,6 @@ def get_waitinglist_for_book(book_key):
         return WaitingLoan.query(identifier=book.ocaid)
     else:
         return []
-    # wl = _query_values(name="book", value=book_key)
-    # # sort the waiting list by timestamp
-    # return sorted(wl, key=lambda doc: doc['since'])
 
 def get_waitinglist_size(book_key):
```
Comment (hornc) [inline openlibrary/core/waitinglist.py]: does this need to be updated to reflect the new changes, or is this not used?

**Comment by hornc** [comment]
Taking a look now... 👀 

 there are tests for the models in openlibrary/tests/core/test_models.py

https://github.com/internetarchive/openlibrary/blob/master/openlibrary/tests/core/test_models.py#L54
should be updated to use the non-deprecated version
`assert e.in_borrowable_collection()`

similarly #L64 
`assert not e.in_borrowable_collection()`

Other things I notice:
openlibrary/tests/core/test_waitinglist.py  has 3 out of 4 tests that aren't doing anything

and core/lending.py has no tests currently, and I believe that is doing most of the heavy lifting. We should add tests for at least all the public methods there.

There seems that lending.py is replacing some of the functionality of waitinglist.py and maybe other older code. There might be an opportunity to prune the older stuff too.

**Comment by hornc** [comment]
basic approach for testing v2 availability:

* Given the various scenarios we want to query IA, confirm we are generating the correct endpoint urls to get the data we need. (and they'll be clearly documented in the tests)
* Mock the responses we expect from IA and test we make the correct decisions based on that data (again, the data we expect from IA is documented, if something changes it should be easy to spot the diffs)

**Comment by mekarpeles** [comment]
Merging this code, retroactively will add tests as we implement our test plan https://github.com/internetarchive/openlibrary/wiki/Testing-Critical-Paths

**Pull request merged**